### PR TITLE
(BKR-524) Add support for the archlinux platform

### PR DESCRIPTION
**Pull request by petems**
* Rebased from the original work by @jantman
* Allows testing of archlinux platform with Beaker

### CODE REVIEW

**Comment by petems** [comment]
Tested successfully with the following SUT:

```
HOSTS:
  archlinux-2016.02.02-amd64:
    roles:
      - master
    platform: archlinux-2016.02.02-amd64
    box: terrywang/archlinux
    box_version: 1.0.0
    hypervisor: vagrant
CONFIG:
  log_level: verbose
  type: aio
```

File: lib/beaker/host/unix/exec.rb [hunk 1]
```diff
@@ -212,6 +212,8 @@ def ssh_permit_user_environment
       exec(Beaker::Command.new("mv #{directory}/sshd_config.permit /etc/ssh/sshd_config"))
     when /(free|open)bsd/
       exec(Beaker::Command.new("sudo perl -pi -e 's/^#?PermitUserEnvironment no/PermitUserEnvironment yes/' /etc/ssh/sshd_config"), {:pty => true} )
+    when /(free|open)bsd/
+      exec(Beaker::Command.new("sudo perl -pi -e 's/^#?PermitUserEnvironment no/PermitUserEnvironment yes/' /etc/ssh/sshd_config"), {:pty => true} )
```
Comment (kevpl) [inline lib/beaker/host/unix/exec.rb]: @petems are these lines different than the ones above them?

**Comment by kevpl** [comment]
@petems would you add a small doc on this platform to our [doc platforms dir](https://github.com/puppetlabs/beaker/tree/master/docs/how_to/hosts), summarizing the particulars of this code (like how since arch is a rolling release, only the latest puppet package versions are supported)?

File: lib/beaker/host/unix/exec.rb (hunk 1 above)
Comment (petems) [inline lib/beaker/host/unix/exec.rb]: Looks like got caught up in a rebase, removed

**Comment by kevpl** [comment]
@puppetlabs-jenkins retest this please

**Comment by kevpl** [comment]
@puppetlabs-jenkins retest this please

**Comment by petems** [comment]
Awesome 👍

**Pull request merged**